main/states/busy_state.py

```diff
@@ -13,16 +13,26 @@ class BusyState(State):
     """Busy state inherits from base class State. In this state, SUSI API is called to perform query and the response
     is then spoken with the selected Text to Speech Service.
     """
-    def detection(self):
+    def detection(self ):
         """This callback is fired when a Hotword Detector detects a hotword.
         :return: None
         """
         # subprocess.call(['killall', 'play'])
         # subprocess.call(['killall', 'mpv']
-        os.system('kill STOP ' + str(self.video_pid))  # nosec #pylint-disable type: ignore
-        subprocess.call(['play', str(self.components.config['detection_bell_sound'])])  # nosec #pylint-disable type: ignore
-        self.transition(self.allowedStateTransitions.get('recognizing'))
-        os.system('kill CONT ' + str(self.video_pid))  # nosec #pylint-disable type: ignore
+        if 'video_pid' in self.__dict__:
```

Comment:
Better:

```py
if getattr(self, `video_pid`):
```

Avoiding abusing `__` functions.

Reply:
@hongquan , on using this approach , I am getting an error `'BusyState' object has no attribute 'audio_pid'`
>if getattr(self, `video_pid`):

Reply:
Show me the code and the log of that error.

You haven't assigned `audio_pid`, so of course `no attribute 'audio_pid'`.

Reply:
Sorry, I remember wrong the function. Should be `hasattr`, not `getattr`.